src/main/java/uk/gov/pay/api/resources/DirectDebitEventsResource.java

```diff
@@ -0,0 +1,109 @@
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Optional;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+@Path("/")
+@Api(value = "/", description = "Public Api Endpoint to get Direct Debit Events")
+@Produces({"application/json"})
+public class DirectDebitEventsResource {
+
+    private final Client client;
+    private final ConnectorUriGenerator connectorUriGenerator;
+
+    @Inject
+    public DirectDebitEventsResource(Client client, ConnectorUriGenerator connectorUriGenerator) {
+        this.client = client;
+        this.connectorUriGenerator = connectorUriGenerator;
+    }
+
+    @GET
+    @Timed
+    @Path("/v1/events")
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(
+            value = "Get direct debit events",
+            notes = "The Authorisation token needs to be specified in the 'authorization' header " +
+                    "as 'authorization: Bearer YOUR_API_KEY_HERE'")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK", response = List.class),
+            @ApiResponse(code = 401, message = "Credentials are required to access this resource")})
+    public Response getDirectDebitEvents(
+            @ApiParam(value = "accountId", hidden = true) @Auth Account account,
+            @QueryParam("before") String beforeDate,
+            @QueryParam("after") String afterDate,
+            @ApiParam(value = "Defaults to a maximum of 500", hidden = false)
+            @QueryParam("page_size") Integer pageSize,
+            @QueryParam("page") Integer page,
+            @ApiParam(value = "ID of associated agreement", hidden = false)
+            @QueryParam("agreement_id") String agreementId,
+            @ApiParam(value = "ID of associated payment", hidden = false)
+            @QueryParam("payment_id") String paymentId
+    ) {
+
+        Optional<BeforeAndAfter> validatedDates = isValid(beforeDate, afterDate);
+        if (!validatedDates.isPresent())
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity("The supplied dates were not in the format yyyy-MM-ddThh:mm:ssZ.")
+                    .build();
+
+        String uri = connectorUriGenerator.eventsURI(account, validatedDates.get().before, validatedDates.get().after, page, pageSize, agreementId, paymentId);
+        Response ddConnectorResponse = client.target(uri)
+                .request()
+                .accept(MediaType.APPLICATION_JSON)
+                .get();
+        
+        if (ddConnectorResponse.getStatus() == HttpStatus.SC_OK) {
+            DirectDebitEventsResponse eventsResponse = ddConnectorResponse.readEntity(DirectDebitEventsResponse.class);
+            return Response.ok(eventsResponse).build();
+        }
+        
+        throw new ConnectorResponseErrorException(ddConnectorResponse);
+    }
+
+    private Optional<BeforeAndAfter> isValid(String beforeDate, String afterDate) {
```

Comment:
This logic here is similar to but subtly different from that in `PaymentSearchValidator.validateDate(String)`

Reply:
Ok I've created a new validator for this purpose and factored the date validation into a new DateValidator class as per convention with MaxLengthValidator and URLValidator.

Reply:
Can this private method go now?